batch/batch/batch.py

```diff
@@ -147,27 +150,116 @@ def batch_record_to_dict(app, record):
         instance.adjust_free_cores_in_memory(rv['delta_cores_mcpu'])
 
 
-def job_record_to_dict(app, record, running_status=None):
-    spec = json.loads(record['spec'])
+async def _get_full_job_spec(app, record):
+    db = app['db']
+    log_store = app['log_store']
+
+    batch_id = record['batch_id']
+    job_id = record['job_id']
+    format_version = record['format_version']
+
+    if format_version == 1:
+        return record['spec']
```

Comment:
If we expect more versions (I expect we do), it be easier to document the various format version checks, extend them, and find them, if we create a FormatVersion class that wraps the version and carries predicates about properties of the format version.  For example:

```
class BatchFormatVersion:
  def __init__(self, format_version):
    self.format_version = format_version

  def spec_in_database(self):
    return self.format_version == 1
```

Then we would never have direct comparisons against the format version.